File: CPP/167.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n = nums.size();
        vector<int> res(2);
        for( int i = 0; i<n; i++){
            
            int temp = target - nums[i];
            
            int l=i+1, r = n-1;
            
            while(l<=r){
                int mid = l+(r-l)/2;
                
                if(nums[mid] == temp){
                    res[0] = i+1;
                    res[1] = mid+1;
                    return res;
                }
                    
                
                else if(nums[mid]<temp)
                    l = mid+1;
                else
                    r=mid-1;
            }
        }
        return res;
    }
};

File: Py/167.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        for i in range(len(numbers)):
            temp = target - numbers[i]
            
            l = i+1
            r = len(numbers)-1
            while l<=r:
                mid = l + (r-l)//2
                
                if numbers[mid] == temp:
                    return [i+1, mid+1]
                elif numbers[mid]>temp:
                    r = mid-1
                else:
                    l = mid+1